GCS/probe0/probe0.go

```diff
@@ -0,0 +1,143 @@
+package probe0
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"io"
+	"os"
+	"time"
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/google/cloudprober/metrics"
+	"github.com/google/cloudprober/probes/options"
+	"github.com/google/cloudprober/targets/endpoint"
+	"github.com/googleinterns/step224-2020/blob/evan/hermesMain/cmd"
+)
+
+// Probe holds aggregate information about all probe runs, per-target.
+type HermesProbe struct {
+	name    string
+	c       *ProbeConf
+	targets []string
+	opts    *options.Options
+
+	res map[string]*metrics.EventMetrics  // Results by target
+	l   *logger.Logger
+}
+
+// Init initializes the probe with the given params.
+func (p *HermesProbe) Init(name string, opts *options.Options) error {
+	p.c, ok := opts.ProbeConf.(*ProbeConf)
+	if !ok {
+		return fmt.Errorf("Invalid Hermes probe configuration")
```

Comment:
I would include an explanation why it was deemed invalid:

fmt.Errorf("invalid argument: the type of opts.ProbeConf is not *ProbeConf") ...or something to that effect.

Also note that error messages should start with lower-case.